hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/GenericRecordTest.java

```diff
@@ -41,6 +58,108 @@
+                .setLong("l", mainDTO.l)
+                .setFloat("f", mainDTO.f)
+                .setDouble("d", mainDTO.d)
+                .setString("str", mainDTO.str)
+                .setGenericRecord("p", innerRecord)
+                .setDecimal("bigDecimal", mainDTO.bigDecimal)
+                .setTime("localTime", mainDTO.localTime)
+                .setDate("localDate", mainDTO.localDate)
+                .setTimestamp("localDateTime", mainDTO.localDateTime)
+                .setTimestampWithTimezone("offsetDateTime", mainDTO.offsetDateTime)
+                .build();
+    }
+
+    @Nonnull
+    private MainDTO createMainDTO() {
+        NamedDTO[] nn = new NamedDTO[2];
+        nn[0] = new NamedDTO("name", 123);
+        nn[1] = new NamedDTO("name", 123);
+        InnerDTO inner = new InnerDTO(new byte[]{0, 1, 2}, new char[]{'c', 'h', 'a', 'r'},
+                new short[]{3, 4, 5}, new int[]{9, 8, 7, 6}, new long[]{0, 1, 5, 7, 9, 11},
+                new float[]{0.6543f, -3.56f, 45.67f}, new double[]{456.456, 789.789, 321.321}, nn,
+                new BigDecimal[]{new BigDecimal("12345"), new BigDecimal("123456")},
+                new LocalTime[]{LocalTime.now(), LocalTime.now()},
+                new LocalDate[]{LocalDate.now(), LocalDate.now()},
+                new LocalDateTime[]{LocalDateTime.now()},
+                new OffsetDateTime[]{OffsetDateTime.now()});
+
+        return new MainDTO((byte) 113, true, 'x', (short) -500, 56789, -50992225L, 900.5678f,
+                -897543.3678909d, "this is main object created for testing!", inner,
+                new BigDecimal("12312313"), LocalTime.now(), LocalDate.now(), LocalDateTime.now(), OffsetDateTime.now());
+    }
+
+
+    @Test
+    public void testGenericRecordToStringValidJson() throws IOException {
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.setEnabled(true);
+        InternalSerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
+
+        MainDTO expectedDTO = createMainDTO();
+        Data data = serializationService.toData(expectedDTO);
+        assertTrue(data.isCompact());
+
+        //internal generic record created on the servers on query
+        InternalGenericRecord internalGenericRecord = serializationService.readAsInternalGenericRecord(data);
+        Json.parse(internalGenericRecord.toString());
+
+        //generic record read from a remote instance without classes on the classpath
+        List<String> excludes = Collections.singletonList("example.serialization");
+        FilteringClassLoader classLoader = new FilteringClassLoader(excludes, null);
+        Thread.currentThread().setContextClassLoader(classLoader);
```

Comment:
Do you need to revert back the class loader for the thread after this test case finishes?

Reply:
Makes sense to be cautious. I will address this.